### PR TITLE
Apb 7795

### PR DESCRIPTION
**Pull request by geoffreywatson**

### CODE REVIEW

File: app/uk/gov/hmrc/agentservicesaccount/controllers/amls/EnterRenewalDateController.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  def onSubmit: Action[AnyContent] = actions.authActionCheckSuspend.async { implicit request =>
+    actions.ifFeatureEnabled(appConfig.enableNonHmrcSupervisoryBody) {
+        withUpdateAmlsJourney { amlsJourney =>
+            renewalDateForm
+            .bindFromRequest()
+            .fold(
+              formWithError => Ok(enterRenewalDate(formWithError)).toFuture,
+              data =>
+                saveAmlsJourney(amlsJourney.copy(newExpirationDate = Option(data))).map(_ =>
+                    Redirect("/page-not-implemented") //TODO - routing
```
Comment (jack-reed) [inline app/uk/gov/hmrc/agentservicesaccount/controllers/amls/EnterRenewalDateController.scala:64]: Do you think we should put a separate ticket in to do all the routing at the end?

File: test/uk/gov/hmrc/agentservicesaccount/views/pages/amls/EnterRenewalDateViewSpec.scala [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        doc.select(".hmrc-header__service-name").first.text() mustBe "Agent services account"
+        doc.select(".hmrc-header__service-name").first.attr("href") mustBe "/agent-services-account"
+      }
+      "have the correct sign out link" in {
+        doc.select(".hmrc-sign-out-nav__link").first.text() mustBe "Sign out"
+        doc.select(".hmrc-sign-out-nav__link").first.attr("href") mustBe "/agent-services-account/sign-out"
+      }
+      "have the correct back link" in {
+        doc.select(".govuk-back-link").first.text() mustBe "Back"
+        doc.select(".govuk-back-link").first.attr("href") mustBe "/todo/not-implemented"
```
Comment (jack-reed) [inline test/uk/gov/hmrc/agentservicesaccount/views/pages/amls/EnterRenewalDateViewSpec.scala:59]: Does this need a TODO comment?

**Review by jack-reed — APPROVED** [review]
I've added some comments but looks good.

File: app/uk/gov/hmrc/agentservicesaccount/controllers/amls/EnterRenewalDateController.scala (hunk 1 above)
Comment (geoffreywatson) [reply app/uk/gov/hmrc/agentservicesaccount/controllers/amls/EnterRenewalDateController.scala:64]: yes...working on that now.

File: test/uk/gov/hmrc/agentservicesaccount/views/pages/amls/EnterRenewalDateViewSpec.scala (hunk 2 above)
Comment (geoffreywatson) [reply test/uk/gov/hmrc/agentservicesaccount/views/pages/amls/EnterRenewalDateViewSpec.scala:59]: that's just a placeholder url...will change with routing logic piece

**Pull request merged**